brainpad/courses/vb/variables_and_nested_statements.md

```diff
@@ -1 +1,472 @@
+
+Time to go back and have fun with our favorite green light! This example will turn the green light on for a specific time, set by a variable called delay. It then turns the light off for the same length of time. Pressing the up and down buttons control the delay variable, causing the light to be on and off for longer or shorter periods. The variable is also printed out using `DebugOutput()`.
+
+```
+Class Program
+
+    Dim delay As Double
+
+    Public Sub BrainPadSetup()
+        delay = 0.2
+    End Sub
+
+    Public Sub BrainPadLoop()
+        If BrainPad.Buttons.IsPressed(BrainPad.Buttons.Button.Up) Then
+            delay = delay + 0.2
+        End If
+        If BrainPad.Buttons.IsPressed(BrainPad.Buttons.Button.Down) Then
+            delay = delay - 0.2
+        End If
+
+        BrainPad.WriteDebugMessage(delay)
+
+        BrainPad.LightBulb.TurnGreen()
+        BrainPad.Wait.Seconds(delay)
+        BrainPad.LightBulb.TurnOff()
+        BrainPad.Wait.Seconds(delay)
+    End Sub
+
+End Class
+
+```
+## Returned Values
+Some methods return a value, like when reading the light sensor. This value can be placed into a variable and then the variable can be used with if-statements. We have lights bulb and a light sensor on the BrainPad. Why not turn the light on when it is dark? This can be tested by turning the lights off inside the room or by placing a finger over the light sensor to block the light.
+```
+Class Program
+
+    Dim level As Double
+
+    Public Sub BrainPadSetup()
+        level = 0
+    End Sub
+
+    Public Sub BrainPadLoop()
+        level = BrainPad.LightSensor.ReadLightLevel()
+        BrainPad.WriteDebugMessage(level)
+
+        If level > 0.5 Then
+            BrainPad.LightBulb.TurnOn()
+        Else
+            BrainPad.LightBulb.TurnOff()
+        End If
+    End Sub
+
+End Class
+```
+How do we know what value to use as the threshold on when to turn the light bulb on? This is a perfect example on where inspecting variables is needed. Run the program and test by placing a finger over the light sensor to block the light. This may not work nicely and you need to adjust the threshold from 0.5 to something else. Take a look at the output window to see the light levels to determine what value work best for your needs. Then stop the program, change the threshold and then run the program again to test the results.
+
+## Exercise
+Modify the last program to have a way to set the threshold using the up button. Hint, the threshold was a constant before but now it is going to be a variable that is set to whatever the light sensor returns when the button is pressed. Name the variable threshold and make sure it is type Double.
+
+## Strings
+A string is a type of variable that holds text. 
+
+```
+BrainPad.WriteDebugMessage("Hello World!")
+```
+It is important to remember that the value 2 and the string "2" are not the same thing. This maybe the same for us but internally in the system they are very different.
```

Comment:
"that the integer 2 and the string "2"" are not the same thing may be more clear.